GetUser/handler.ts

```diff
@@ -219,51 +218,53 @@ export function GetUserHandler(
                   "Could not get user by email.",
                   error as Error
                 )
-            ).map(userList => userList[0])
-          )
-          .map(adb2User => ({
+            )
+          ),
+          TE.map(([adb2User]) => ({
             ...taskResults,
             token_name: adb2User[`${adb2cTokenAttributeName}`]
           }))
-      )
-      .chain(
-        fromPredicate(
-          userInfo => isRight(userInfo.errorOrGroups),
-          userInfoWithError =>
-            internalErrorHandler(
-              "Invalid group contract from APIM.",
-              userInfoWithError.errorOrGroups.value as Error
-            )
         )
-      )
-      .chain(
-        fromPredicate(
-          userInfo => isRight(userInfo.errorOrSubscriptions),
-          userInfoWithError =>
-            internalErrorHandler(
-              "Invalid subscription contract from APIM.",
-              userInfoWithError.errorOrSubscriptions.value as Error
+      ),
+      TE.chain(userInfo =>
+        E.isRight(userInfo.errorOrGroups)
+          ? TE.of(userInfo)
+          : TE.left(
+              internalErrorHandler(
+                "Invalid group contract from APIM.",
+                userInfo.errorOrGroups.left
+              )
```

Comment:
We can use `TE.fromPredicate`

Reply:
Not that easy: the subject of the predicate is `userInfo`, but the `isRight` guard refers to `userInfo.errorOrGroups`.

That is, to use `fromPredicate` will result in the following:
```ts
fromPredicate(
	userInfo => E.isRight(userInfo.errorOrGroups),
    //     ^^^ the return type is a guard on userInfo, which tells nothing about userInfo.errorOrGroups
    () => 
		internalErrorHandler(
        	"Invalid group contract from APIM.",
            userInfo.errorOrGroups.left
			//                     ^^^^ error: errorOrGroups isn't narrowed to `Left<T>`, thus left property cannot be accessed
        )
)

```